source/common/network/socket_option_factory.h

```diff
@@ -32,6 +32,7 @@ class SocketOptionFactory : Logger::Loggable<Logger::Id::connection> {
   static std::unique_ptr<Socket::Options> buildIpPacketInfoOptions();
   static std::unique_ptr<Socket::Options> buildRxQueueOverFlowOptions();
   static std::unique_ptr<Socket::Options> buildReusePortOptions();
+  static std::unique_ptr<Socket::Option> buildSocketLingerOption(int onoff, int timeout);
```

Comment:
Can you make the parameters here a bit more platform-agnostic? We don't care about the underlying unix parameter details AFAICT and only won't a very specific behavior. This will more easily allow this to be ported to NT in the future. We can always add more parameters later as needed.

Reply:
Something like `buildSocketNoLingerOption()` ??

Reply:
Yes that would be fine.